### PR TITLE
[OE] Create Connection with new APIs of CE

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary
Create the connection between `Orchestration Engine` and `Consent Engine` with the new APIs implemented at the `Consent Engine`

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [x] Performance improvement
- [ ] Other (please describe):

## Changes Made
- Consolidate consent client logic into CEServiceClient and remove legacy code
- Add MultiConsentResponse structure and update CreateConsentRecord to return multiple consent records
- Update consent client to CEServiceClient and enhance consent handling logic

## Testing
- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues
Related to #337

### CODE REVIEW

File: exchange/orchestration-engine/consent/ce_client.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		return nil, err
+	}
+
+	return &consentResponse, nil
+}
+
+// GetConsent retrieves a consent record by (OwnerID, AppID) or (OwnerEmail, AppID)
+func (c *CEServiceClient) GetConsent(ownerID *string, ownerEmail *string, appID string) (*ConsentResponseInternalView, error) {
+	// Implementation of the method to send HTTP request to get consent
+	req, err := http.NewRequest("GET", c.baseURL+"/consents", nil)
```
Comment (mushrafmim) [inline exchange/orchestration-engine/consent/ce_client.go]:
```suggestion
	req, err := http.NewRequest(http.MethodGet, c.baseURL+"/consents", nil)
```

File: exchange/orchestration-engine/federator/federator.go [hunk 2]
```diff
@@ -376,33 +374,26 @@ func (f *Federator) FederateQuery(ctx context.Context, request graphql.Request,
...
-				},
-			},
-		}
-
-		ceResp, err := ceClient.MakeConsentRequest(ceRequest)
+		// Check if there is already existing consent for the required fields
+		// For now we assume consent is identified by citizen ID (extractedArgs[0])
+		// And the consent is required from only one owner (citizen)
+		// GetConsent is expected to return the most recently created consent record for the given OwnerEmail and AppID.
+		ownerEmail := func() string {
```
Comment (ginaxu1) [inline exchange/orchestration-engine/federator/federator.go]: nit: `ownerEmail` can be empty string here and still used in operations (can cause unexpected behavior in CE). suggest adding validation that ownerEmail is non-empty before use


```go
ownerEmail := func() (string, error) {
    val := extractedArgs[0].Value.GetValue()
    if s, ok := val.(string); ok && s != "" {
        return s, nil
    }
    return "", fmt.Errorf("ownerEmail is not a valid non-empty string, got: %v", val)
}()
if ownerEmail == "" {
    return graphql.Response{
        Data: nil,
        Errors: []interface{}{
            map[string]interface{}{
                "message": "Invalid owner email",
                "extensions": map[string]interface{}{
                    "code": errors.CodeMissingEntityIdentifier,
                },
            },
        },
    }
}
```

File: exchange/orchestration-engine/consent/ce_client.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	"net/http"
+	"time"
+
+	"github.com/ginaxu1/gov-dx-sandbox/exchange/orchestration-engine/logger"
+)
+
+// CEServiceClient represents a client to interact with the Consent Engine service
+type CEServiceClient struct {
+	baseURL    string
+	HTTPClient *http.Client
```
Comment (ginaxu1) [inline exchange/orchestration-engine/consent/ce_client.go]: `HTTPClient` is inconsistent with Go naming conventions, use lowercase for unexported field
```go
type CEServiceClient struct {
    baseURL    string
    httpClient *http.Client 
}
```

File: exchange/orchestration-engine/consent/ce_client.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+func (c *CEServiceClient) CreateConsent(request *CreateConsentRequest) (*MultiConsentResponse, error) {
+	// Implementation of the method to send HTTP request to create consent
+	requestBody, err := json.Marshal(request)
+	if err != nil {
+		logger.Log.Error("Failed to marshal CreateConsentRequest", "error", err)
+		return nil, err
+	}
+
+	logger.Log.Info("Making Create Consent Request to Consent Engine", "url", c.baseURL+"/consents")
+	req, err := http.NewRequest("POST", c.baseURL+"/consents", bytes.NewBuffer(requestBody))
```
Comment (ginaxu1) [inline exchange/orchestration-engine/consent/ce_client.go]: nit: L37 + 67 in `exchange/orchestration-engine/consent/ce_client.go`
Use constants for string literals that appear multiple times 


```go
const (
    consentEndpoint = "/consents"
    httpMethodPOST  = "POST"
    httpMethodGET   = "GET"
)
```
